bpf/flows_filter.h

```diff
@@ -0,0 +1,240 @@
+                        } else {
+                            result = 0;
+                            goto end;
+                        }
+                    }
+                    // srcPort matching
+                    if (rule->srcPortStart != 0 && rule->srcPortEnd == 0) {
+                        if (rule->srcPortStart == id->src_port) {
+                            BPF_PRINTK("srcPortStart matched\n");
+                            result++;
+                        } else {
+                            result = 0;
+                            goto end;
+                        }
+                    } else if (rule->srcPortStart != 0 && rule->srcPortEnd != 0) {
+                        if (rule->srcPortStart <= id->src_port && id->src_port <= rule->srcPortEnd) {
+                            BPF_PRINTK("srcPortStart and srcPortEnd matched\n");
+                            result++;
+                        } else {
+                            result = 0;
+                            goto end;
+                        }
+                    }
+                    // Generic port matching check for either src or dst port
+                    if (rule->portStart != 0 && rule->portEnd == 0) {
+                        if (rule->portStart == id->src_port || rule->portStart == id->dst_port) {
+                            BPF_PRINTK("portStart matched\n");
+                            result++;
+                        } else {
+                            result = 0;
+                            goto end;
+                        }
+                    } else if (rule->portStart != 0 && rule->portEnd != 0) {
+                        if ((rule->portStart <= id->src_port && id->src_port <= rule->portEnd) ||
+                            (rule->portStart <= id->dst_port && id->dst_port <= rule->portEnd)) {
+                            BPF_PRINTK("portStart and portEnd matched\n");
+                            result++;
+                        } else {
+                            result = 0;
+                            goto end;
+                        }
+                    }
+                    break;
+                case IPPROTO_ICMP:
+                case IPPROTO_ICMPV6:
+                    if (rule->icmpType != 0) {
+                        if (rule->icmpType == id->icmp_type) {
+                            BPF_PRINTK("icmpType matched\n");
+                            result++;
+                        } else {
+                            result = 0;
+                            goto end;
+                        }
+                    }
+                    if (rule->icmpCode != 0) {
+                        if (rule->icmpCode == id->icmp_code) {
+                            BPF_PRINTK("icmpCode matched\n");
+                            result++;
+                        } else {
+                            result = 0;
+                            goto end;
+                        }
+                    }
```

Comment:
I don't think it makes sense to have `icmpCode` independent of `icmpType` so we should move that check inside of `if (rule->icmpType != 0)` condition above and document that `icmpCode` requires `icmpType` to be set.

WDYT ?

Reply:
they  are two independent filters so u set one or the other or both or none

Reply:
yes, but each type have its own related code implementation:
https://www.iana.org/assignments/icmp-parameters/icmp-parameters.xhtml

so it doesn't make any sense to have `icmpCode` filter only